lua/weapons/weapon_golfclub/shared.lua

```diff
@@ -211,268 +208,269 @@ function SWEP:SetupDataTables()
+		local record = 10
+		if SERVER then
+			record = tonumber(self:GetOwner():GetPData("golf_"..hole, 10))
+			monthlyrecord = tonumber(self:GetOwner():GetPData("monthly_golf_"..hole, 10))
+			local curdate = os.date("*t",os.time())
+			curdate = curdate["year"]*12 + curdate["month"]
+			if curdate > tonumber(self:GetOwner():GetPData("monthly_reset",0)) then
+				for h,_ in pairs(GolfPrizeTargets) do
+					self:GetOwner():SetPData("monthly_golf_"..h,10)
 				end
+				self:GetOwner():SetPData("monthly_reset",curdate)
 			end
-		end
-	else
-		local strokes = " strokes"
-		if record==1 then strokes = " stroke" end
-		self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]Your record for this hole is: [orange]"..tostring(record)..strokes )
-	end
-	local t = GolfPrizeTargets[hole]
-	if t then
-		local got = math.min(strk,record)
-		local nex = nil
-		for k,v in pairs(t) do
-			if got > v[1] then
-			if nex == nil or nex[1]<v[1] then
-				nex = v
+			if strk<monthlyrecord then
+				self:GetOwner():SetPData("monthly_golf_"..hole, tostring(strk))
 			end
+			if strk<record then
+				self:GetOwner():SetPData("golf_"..hole, tostring(strk))
 			end
 		end
-		if nex == nil then
-			if iwonsomething==false then
-				self:GetOwner():PrintMessage(HUD_PRINTTALK, "[red]You've claimed all the prizes for this hole!")
-			end
-		else
+		if !joke then GolfNotify("[yellow]"..self:GetOwner():Nick() .." completed hole '"..hole.. "' with "..endm) end
+		if SERVER then
+			local iwonsomething = false
 			local strokes = " strokes"
-			if nex[1]==1 then strokes = " stroke" end
-			self:GetOwner():PrintMessage(HUD_PRINTTALK, "[red]Beat this hole in [orange]"..tostring(nex[1])..strokes.."[red] to win [gold]"..tostring(nex[2]).." points!")
+			local alltime = false
+			if strk<record then
+				if strk==1 then strokes = " stroke" end
+				self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]You beat your all-time record! New record: [orange]"..tostring(strk)..strokes )
+				alltime = true
+			end
+			strokes = " strokes"
+			if strk<monthlyrecord then
+				if strk==1 then strokes = " stroke" end
+				if not alltime then self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]You beat your monthly record! New record: [orange]"..tostring(strk)..strokes ) end
+				local t = GolfPrizeTargets[hole]
+				if t then
+					for k,v in pairs(t) do
+						if monthlyrecord > v[1] and strk <= v[1] then
+							if self:GetOwner().familyshared then
+								self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]Sorry, you're not eligible for this prize ;hahaha;" )
+							else
+								self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]Congratulations! You won [gold]"..tostring(v[2]).." points!" )
+								self:GetOwner():PS_GivePoints(v[2])
+							end
+							iwonsomething = true
+						end
+					end
+				end
+			else
+				if monthlyrecord==1 then strokes = " stroke" end
+				self:GetOwner():PrintMessage( HUD_PRINTTALK, "[red]Your all-time record for this hole is: [orange]"..tostring(record)..strokes )

```

Comment:
Do this with one line please:
"Your all-time record for this hole is X (X this month)"